openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -81,50 +82,53 @@
  *
  * @since 1.9.0
  */
-public class OpenAPIBuildToolRunner implements BuildToolRunner {
-    static List<Diagnostic> diagnostics = new ArrayList<>();
+public class OpenAPICodeGeneratorTool implements CodeGeneratorTool {
+    static String md5HexOpenAPI;
 
     @Override
-    public String getToolName() {
+    public String toolName() {
         return "openapi";
     }
 
     @Override
-    public List<Diagnostic> diagnostics() {
-        return diagnostics;
-    }
-
-    @Override
-    public void executeTool(ToolContext toolContext) {
-
+    public void execute(ToolContext toolContext) {
         ImmutablePair<OASClientConfig, OASServiceMetadata> codeGeneratorConfig;
-        TomlNodeLocation location = toolContext.packageInstance().ballerinaToml().get().tomlAstNode().location();
+        TomlNodeLocation location = toolContext.currentPackage().ballerinaToml().get().tomlAstNode().location();
         try {
             // Validate the OAS file whether we can handle within OpenAPI tool
             if (!canHandle(toolContext)) {
                 Constants.DiagnosticMessages error = Constants.DiagnosticMessages.WARNING_FOR_UNSUPPORTED_CONTRACT;
-                reportDiagnostics(error, error.getDescription(), location);
+                createDiagnostics(toolContext, error, error.getDescription(), location);
```

Comment:
why do we need to pass both the `error` and `error.getDescription()` as arguments to the same method?
```suggestion
                createDiagnostics(toolContext, error, error.getDescription(), location);
```

Reply:
There are some extra details included to the modified message and that is why we have separate errorMessage, anyhow will try to improve this method by passing varargs